binary_search_tree/tree.py

```diff
@@ -11,48 +11,129 @@ def __init__(self, key, val = None):
 
 
 class Tree:
-    def __init__(self):
-        self.root = None
+    def __init__(self, root=None):
+        self.root = root
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
-
-    # Time Complexity: 
-    # Space Complexity: 
+    def add_helper (self, current, key, value):
+        if current == None:
+            return TreeNode(key, value)
+        elif current.key >= key:
+            current.left = self.add_helper(current.left, key, value)
+        else: 
+            current.right = self.add_helper(current.right, key, value)
+        return current 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(log n)
+    def add(self, key, value=None):
+        if self.root == None:
+            self.root = TreeNode(key, value)
+        else:
+            self.add_helper(self.root, key, value)
+    def find_helper(self, current, key):
```

Comment:
✨ Nice work! We can clean up your code a bit by refactoring to check if `current is None` initially instead of checking the left and right children. The next recursive calls, the left and right children will be `current` and the following conditional statement will catch them!

```suggestion
    def find_helper(self, current, key):
        if current is None:
            return None
```